server/service/apple_mdm.go

```diff
@@ -414,7 +414,11 @@ func listMDMAppleConfigProfilesEndpoint(ctx context.Context, request interface{}
 }
 
 func (svc *Service) ListMDMAppleConfigProfiles(ctx context.Context, teamID uint) ([]*fleet.MDMAppleConfigProfile, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.MDMConfigProfileAuthz{TeamID: &teamID}, fleet.ActionRead); err != nil {
+	action, err := viewer.DetermineActionAllowingGitOps(ctx, fleet.ActionList)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting action for user")
+	}
+	if err := svc.authz.Authorize(ctx, &fleet.MDMConfigProfileAuthz{TeamID: &teamID}, action); err != nil {
```

Comment:
Hey @lucasmrod , sorry to bother, before making this ready for review, I wanted to sanity check and coordinate on this approach to give gitops read access to specific endpoints.

Based on [this conversation](https://github.com/fleetdm/fleet/issues/15337#issuecomment-1932878997) with Noah and Rachael, we don't want to include `for_update` in any future endpoints, so I wanted to check with you about this approach (as you mentioned you have to do something similar for other endpoints)

Reply:
To be useful gitops will need plenty of more access, see https://github.com/fleetdm/fleet/pull/16710.

I think we should be explicit on the rego file. If gitops requires read access let's amend the rego.

Reply:
/cc @getvictor (given you are working on gitops)

Reply:
I wonder if we should introduce new authz structs? Taking this case for example, just adding gitops read access to `fleet.Host` will affect a bunch of other endpoints

Reply:
I just saw Victor's PR, I will go with a different auth struct for host, we can always tweak that later.

Reply:
+1 to adding more granularity (either new entities or actions). I'm happy to review what you have in mind.

Reply:
adding actions is an awesome idea I haven't considered, thanks!